x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/components/host_details_flyout/flyout_wrapper.tsx

```diff
@@ -18,10 +18,10 @@ export interface Props {
 }
 
 export const FlyoutWrapper = ({ node: { name }, closeFlyout }: Props) => {
-  const { source } = useSourceContext();
+  const { metricsView } = useMetricsDataView();
```

Comment:
Is it necessary to get the indexPattern from `useMetricsDataView` when `useSourceContext` has it?  `useMetricsDataView` is already calling `useSourceContext` and then it has to make another request for the data view.

Reply:
This is wrong, it should be using `useMetricsDataViewContext`. 

My idea is to fully deprecate `useSourceContext` for index-related stuff, and move that to `useMetricsDataView`.
The data view object is cached, so unless the user hasn't reloaded the page, no additional requests will be made. 

Also, the data view is first loaded in `metrics_app.tsx` and that's where only requests it should happen. After that ti will retrieve what's cached

Reply:
> My idea is to fully deprecate useSourceContext for index-related stuff, and move that to MetricsDataViewContext.
The data view object is cached, so unless the user hasn't reloaded the page, no additional requests will be made

That sounds fine to me.   Also related to my other comment, I think there could be a scenarios where we want the index pattern and probably would not want to create a data view.  I can think of the infrastructure nodes table that APM is using that [uses this hook](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/metrics_data_access/public/components/infrastructure_node_metrics_tables/shared/hooks/use_infrastructure_node_metrics.ts#L28-L63) which gets the index pattern from data access plugin.   Since we'd need the data view with the flyout when that gets shared, perhaps we'd need to move the data view management there as well.

Reply:
> I can think of the infrastructure nodes table that APM is using that [uses this hook](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/metrics_data_access/public/components/infrastructure_node_metrics_tables/shared/hooks/use_infrastructure_node_metrics.ts#L28-L63) which gets the index pattern from data access plugin. Since we'd need the data view with the flyout when that gets shared, perhaps we'd need to move the data view management there as well.

I agree. I'll see how this can be improved.